sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-verifyAssessmentCalculationOrder.e2e-spec.ts

```diff
@@ -1217,4 +1222,257 @@ describe("AssessmentController(e2e)-verifyAssessmentCalculationOrder", () => {
+          disbursementValues: [
+            createFakeDisbursementValue(
+              DisbursementValueType.CanadaGrant,
+              "CSPT",
+              1150,
+              { disbursedAmountSubtracted: 1000 },
+            ),
+            createFakeDisbursementValue(
+              DisbursementValueType.CanadaGrant,
+              "CSGD",
+              500,
+              { disbursedAmountSubtracted: 500 },
+            ),
+          ],
+        },
+        {
+          initialValues: {
+            coeStatus: COEStatus.required,
+            disbursementScheduleStatus: DisbursementScheduleStatus.Pending,
+          },
+        },
+      );
+      currentAssessment.disbursementSchedules = [currentAssessmentDisbursement];
+      await db.studentAssessment.save(currentAssessment);
+
+      // Application currently being processed.
+      const currentApplication = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        { student },
+        {
+          offeringIntensity: OfferingIntensity.partTime,
+          applicationStatus: ApplicationStatus.InProgress,
+          currentAssessmentInitialValues: {
+            assessmentWorkflowId: "some fake id",
+            studentAssessmentStatus: StudentAssessmentStatus.InProgress,
+          },
+        },
+      );
+
+      // Act
+      const result =
+        await assessmentController.verifyAssessmentCalculationOrder(
+          createFakeVerifyAssessmentCalculationOrderPayload(
+            currentApplication.currentAssessment.id,
+          ),
+        );
+
+      // Asserts
+      expect(FakeWorkerJobResult.getResultType(result)).toBe(
+        MockedZeebeJobResult.Complete,
+      );
+      // Expect to consider only the CSGP from the first disbursement
+      // and ignore the second disbursement with the declined COE.
+      expect(FakeWorkerJobResult.getOutputVariables(result)).toStrictEqual({
+        isReadyForCalculation: true,
+        latestCSLPBalance: 0,
+        programYearTotalPartTimeCSPT: 1150,
+        programYearTotalPartTimeCSGD: 500,
+      });
+    },
+  );
+
+  it(
+    "Should consider the sum of grants only from the sent disbursement when a past application has COE declined for current assessment disbursement" +
+      " but has a disbursement sent from previous assessment.",
+    async () => {
+      // Arrange
+
+      // Create the student to be shared across the applications.
+      const student = await saveFakeStudent(db.dataSource);
+      // Past part-time application with two assessments where the first assessment disbursement is sent
+      // and current assessment disbursement COE is declined and the disbursement is cancelled.
+      const pastApplication = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        {
+          student,
+          firstDisbursementValues: [
+            createFakeDisbursementValue(
+              DisbursementValueType.CanadaGrant,
+              "CSPT",
+              1000,
```

Comment:
Similar to this one: https://github.com/bcgov/SIMS/pull/4522/files#r2010893134
I would say the `valueAmount` and `effectiveAmount` could differ even if by one to ensure the result is asserted accurately (even if there is a deviation from a real-world scenario).